packages/web3-common/src/deferred_promise.ts

```diff
@@ -0,0 +1,88 @@
+import { OperationTimeoutError } from './errors';
+
+export class DeferredPromise<T> implements Promise<T> {
+	public [Symbol.toStringTag]: 'Promise';
```

Comment:
Whenever you do something obscure like this you should add a comment explaining why for not only ourselves, but future devs as well